samples/net/sockets/http_server/src/not_found_page.html

```diff
@@ -1,10 +0,0 @@
-<!DOCTYPE html>
```

Comment:
Why the not found page was removed?

Reply:
I couldn't see the generated `not_found_page.html.gz.inc` file referenced anywhere in the sample.

I've now gone back and had another look, and it seems the file is referenced in `zephyr/subsys/net/lib/http/http_server_http2.c`, but only if `INCLUDE_HTML_CONTENT` is defined. This currently isn't defined anywhere (there's a CMake option `INCLUDE_HTML_CONTENT` but this doesn't set the preprocessor definition)
```
static const char content_404[] = {
#ifdef INCLUDE_HTML_CONTENT
#include "not_found_page.html.gz.inc"
#endif
};
```

Reply:
Ok, some piece of the puzzle is now missing, I think this used to work earlier. Anyway, either we remove the "not found" file + code in http2, or this needs to be fixed properly, but both cases should be done outside of this pr.

Reply:
Maybe a Kconfig option in the HTTP server lib to specify a file path to the user's custom 404 page might be clearer? A bit like how `CONFIG_MCUBOOT_SIGNATURE_KEY_FILE` works.

Reply:
One of the awkwardness we've left out from the initial server code, yes I think it'd be better to have this configurable with Kconfig.